src/System.Runtime/tests/System/DateTimeTests.cs

```diff
@@ -2177,6 +2178,59 @@ public void GetObjectData_NullInfo_ThrowsArgumentNullException()
             AssertExtensions.Throws<ArgumentNullException>("info", () => ((ISerializable)DateTime.Now).GetObjectData(null, new StreamingContext()));
         }
 
+        [Fact]
+        public void TestRoundTrippingDateTimeAndFileTime()
+        {
+            // This test ensure the round tripping of DateTime with the system file time.
+            // It is important to have this working on systems support leap seconds as the conversion wouldn't be simple
+            // conversion but involve some OS calls to ensure the right conversion is happening.
+
+            DateTime now = DateTime.UtcNow;
+            long fileTime = now.ToFileTimeUtc();
+            DateTime roundTrippedDateTime = DateTime.FromFileTimeUtc(fileTime);
+            Assert.Equal(now, roundTrippedDateTime);
+
+            now = DateTime.Now;
+            fileTime = now.ToFileTime();
+            roundTrippedDateTime = DateTime.FromFileTime(fileTime);
+            Assert.Equal(now, roundTrippedDateTime);
+        }
```

Comment:
Do we need a similar test for DateTimeOffset?

Reply:
DateTimeOffset is using DateTime for such operations.

Reply:
We already have similar test for DateTimeOffset https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/DateTimeOffsetTests.cs#L518